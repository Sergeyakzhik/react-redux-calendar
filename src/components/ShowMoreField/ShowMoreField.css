.show-more {
  position: absolute;
  background-color: #000000ed;
  width: 38rem;
  left: 50%;
  margin-left: -19rem;
  border-radius: 10px;
  padding: 1rem;
  z-index: 2000;
  top: 10rem;
  color: white;
}

.show-more h1 {
  margin-bottom: 2rem;
}

.show-more li {
  margin-left: 1rem;
  padding-right: 2rem;
  width: 100%;
  margin-bottom: 1rem;
}

.show-more h5 {
  word-break: break-word;
}

.show-more hr {
  border: 1px solid #7b7b7b;
  width: 31rem;
  margin-right: 4rem;
}
