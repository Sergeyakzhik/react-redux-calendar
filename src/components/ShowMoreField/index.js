import React from 'react';
import './style.css';

const ShowMoreField = props => (
  <div className="show-more">
    <div className="close-button" onClick={props.onCloseButtonClick}>
      <i className="fas fa-times" />
    </div>
    <div>
      <ul>
        {props.events.map(event => (
            <>
              <hr />
              <li>
                <div className="row">
                  <div className="col-sm-6">
                    <h3>{event.name}</h3>
                    <h4>{`${event.startDate.format('h:mm a')} - ${event.endDate.format('h:mm a')}`}</h4>
                    <h4>{event.place}</h4>
                  </div>
                  <div className="col-sm-6">
                    <h6>{event.description || 'No description'}</h6>
                  </div>
                </div>
              </li>
            </>
        ))}
      </ul>
    </div>
  </div>
);

export default ShowMoreField;
