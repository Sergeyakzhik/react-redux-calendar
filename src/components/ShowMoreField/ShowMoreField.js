import React from 'react';
import './ShowMoreField.css';

import { connect } from 'react-redux';

class ShowMoreField extends React.Component {
  render() {
    return (
      <div className="show-more">
        <h1>{this.props.date}</h1>
        <div className="close-button" onClick={this.props.onEndButtonClick}>
          <i className="fas fa-times"></i>
        </div>
        <div>
          {this.props.events.map(event => event.startDate)}
        </div>
      </div>
    );
  }
}

const mapStateToProps = store => ({
  events: store.showMoreField.events
});

export default connect(
  mapStateToProps
)(ShowMoreField);
