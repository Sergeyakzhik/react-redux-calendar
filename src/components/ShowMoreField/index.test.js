import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import moment from 'moment';
import ShowMoreField from './index';


Enzyme.configure({ adapter: new Adapter() });

describe('ShowMoreField', () => { // FIX: ShowMoreField component
  const date = moment();
  const props = {
    events: {
      event1: {
        name: 'new Event',
        startDate: date,
        endDate: date,
      },
      event2: {
        name: 'new Event2',
        startDate: date,
        endDate: date,
      },
    },
  };
  let component;

  beforeEach(() => {
    component = shallow(<ShowMoreField {...props} />);
  });

  it('renders ShowMoreField component', () => {
    expect(component).toMatchSnapshot();
  });

  it('checks ShowMoreField name', () => {
    expect(component.find('h1').text()).toEqual(date.format('MMMM Do YYYY'));
  });
});
