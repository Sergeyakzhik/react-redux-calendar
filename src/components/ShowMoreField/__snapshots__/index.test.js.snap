// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreField render ShowMoreField component 1`] = `undefined`;
